pkg/daemon/outputs.go

```diff
@@ -27,10 +29,21 @@ func (srv *Daemon) outputsHandler(engine api.Engine) func(w http.ResponseWriter,
 
 		tgw := rpc.NewOutputWriter(w, r)
 
-		err = engine.DoCollectOutputs(r.Context(), req.Runner, req.RunID, tgw)
+		result := false
+		defer func() {
+			tgw.WriteResult(result)
+		}()
+
+		var file bytes.Buffer
+		fw := bufio.NewWriter(&file)
+
+		err = engine.DoCollectOutputs(r.Context(), req.Runner, req.RunID, tgw, fw)
```

Comment:
We shouldn't need this, nor the `io.Writer` parameter in `DoCollectOutputs`. You can create a `OutputWriter.BinaryWriter()` that returns an `io.Writer` that wires writes to the underlying `OutputWriter.WriteBinary`. See `OutputWriter.StdoutWriter()` and `OutputWriter.InfoWriter()` for inspiration.